source/elastic_solver.py

```diff
@@ -173,7 +173,8 @@ def _check_boundary_condition_format(self, bc):
         assert len(bc) >= 2
         # 1. check bc type
         assert isinstance(bc[0], (DisplacementBCType, TractionBCType))
-        rank = 0 # To Do: Changed it from rank = 1
+        # To Do: Changed it from rank = 1
```

Comment:
We don't where this otherwise would change to 0. We need this for the pressure bc.

Reply:
Could you specify an example or a test where these lines fail? I can imagine that it might be necessary to introduce an ` enum PressureBCType` to distinguish between the boundary conditions in the incompressible case.

Do you specify any boundary conditions on space of `p` and `q`, respectively?